sdk/go/common/workspace/plugins.go

```diff
@@ -967,6 +968,29 @@ func getHTTPResponse(req *http.Request) (io.ReadCloser, int64, error) {
 	// detailed api logs that may include credentials.
 	logging.V(11).Infof("plugin install request headers: %v", req.Header)
 
+	resp, err := http.DefaultClient.Do(req)
+	if err != nil {
+		return nil, -1, err
+	}
+
+	// As above this might include authentication information, but also to be consistent at what level headers
+	// print at.
+	logging.V(11).Infof("plugin install response headers: %v", resp.Header)
+
+	if resp.StatusCode < 200 || resp.StatusCode > 299 {
+		contract.IgnoreClose(resp.Body)
+		return nil, -1, newDownloadError(resp.StatusCode, req.URL, resp.Header)
+	}
+
+	return resp.Body, resp.ContentLength, nil
+}
+
+func getHTTPResponseWithRetry(req *http.Request) (io.ReadCloser, int64, error) {
```

Comment:
Can we kill this and just use our retry logic in getLatest instead? `DoWithRetry` doesn't give us any way to hook to tell the user that we are retrying.

Reply:
Could you clarify? Are you suggesting independent retry logic in GetLatestVersion or in its call sites?
GetLatestVersion probably shouldn't do reporting directly--unless we pass an output stream to it.